ui.frontend/src/utils.js

```diff
@@ -94,17 +90,35 @@ export default class Utils {
     }
 
     /**
-     * Creates a field view and add form container
+     * Creates fields from given elements of fieldCreator, for given form container
+     * @param fieldElements
      * @param fieldCreator
-     * @param fieldElement
      * @param formContainer
      */
-    static createFormContainerField(fieldCreator, fieldElement, formContainer) {
-        let field = fieldCreator({
-            "element" : fieldElement,
-            "formContainer" : formContainer
+    static createFormContainerFields(fieldElements, fieldCreator, formContainer) {
+        for (let i = 0; i < fieldElements.length; i++) {
+            const elementId = fieldElements[i].id;
+            //check if field is already created, to avoid creating a new field for same id
+            if (formContainer.getField(elementId) == null) {
+                let field = fieldCreator({
+                    "element" : fieldElements[i],
+                    "formContainer" : formContainer
+                });
+                formContainer.addField(field);
+            }
+        }
+    }
+
+    /**
+     * Creates fields from all registered fieldCreators present inside addedElement, for given form container
+     * @param addedElement
+     * @param formContainer
+     */
+    static createFieldsForAddedElement(addedElement, formContainer) {
```

Comment:
if its only used by instance manager, I'd suggest to create a local internal function. this has made the API public. We should only be exposing such APIs on actual use-case/demand.